SplitFiles/bread.c

```diff
@@ -18,10 +18,11 @@
 #include "bread.h"
 #include "util.h"
 
-BreadType*  rand_bread_type(BreadBin* bin) {
+BreadType* rand_bread_type(BreadBin* bin) {
     int i;
+    int rand_choice;
     rand_choice = rand() % bin->probTotal;
-    for (int i = 0; i < BREADBINSIZE; ++i) {
+    for (i = 0; i < BREADBINSIZE; ++i) {
```

Comment:
This picks out a random slice based on rand_choice and the weighting each bread has.  It steps through all available bread types (ie not NULL pointers) until the total weighting equals or exceeds rand_choice.  Note this code doesn't need to know what bread types are available.